File: src/backend/music-server/music-api/src/main/java/com/lalala/music/config/WebConfiguration.java
package com.lalala.music.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebConfiguration implements WebMvcConfigurer {
    private static final Long MAX_AGE = 3000L;

    @Override
    public void addCorsMappings(final CorsRegistry registry) {
        registry
                .addMapping("/**")
                .allowedOrigins("127.0.0.1:3000")
                .allowedHeaders("*")
                .allowCredentials(false)
                .maxAge(MAX_AGE);
    }
}

File: src/backend/chart-server/chart-api/src/main/kotlin/com/lalala/chart/config/WebConfiguration.kt
package com.lalala.chart.config

import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.config.CorsRegistry
import org.springframework.web.reactive.config.WebFluxConfigurer

@Configuration
class WebConfiguration : WebFluxConfigurer {
    override fun addCorsMappings(registry: CorsRegistry) {
        registry.addMapping("/**")
            .allowedOrigins("*")
            .allowedMethods("*")
            .allowedHeaders("*")
            .allowCredentials(false)
            .maxAge(60000)
    }
}
